pandas/_libs/groupby.pyx

```diff
@@ -1286,7 +1286,7 @@ def group_quantile(
                     elif interp == INTERPOLATION_MIDPOINT:
                         out[i, k] = (val + next_val) / 2.0
                     elif interp == INTERPOLATION_NEAREST:
-                        if frac > .5 or (frac == .5 and q_val > .5):  # Always OK?
+                        if frac > .5 or (frac == .5 and idx % 2 == 1):  # Always OK?
```

Comment:
Could you replace `# Always OK?` with a short comment above what `idx % 2 == 1` is for?

Reply:
I added a comment.